app/react/Users/components/usergroups/UserGroups.tsx

```diff
@@ -0,0 +1,100 @@
+import React, { useEffect, useState } from 'react';
+import { connect } from 'react-redux';
+import { IImmutable } from 'shared/types/Immutable';
+import { GroupMemberSchema, UserGroupSchema } from 'shared/types/userGroupType';
+import { IStore } from 'app/istore';
+import { UserGroupList } from 'app/Users/components/usergroups/UserGroupList';
+import {
+  deleteUserGroup,
+  loadUserGroups,
+  saveUserGroup,
+} from 'app/Users/components/usergroups/actions/actions';
+import { UserGroupSidePanel } from './UserGroupSidePanel';
+
+export interface UserGroupProps {
+  userGroups: IImmutable<UserGroupSchema[]>;
+  users: IImmutable<GroupMemberSchema[]>;
+  loadUserGroups: () => any;
+  saveUserGroup: (userGroup: UserGroupSchema) => Promise<void>;
+  deleteUserGroup: (userGroup: UserGroupSchema) => Promise<void>;
+}
+
+function UserGroups({
+  userGroups,
+  users,
+  loadUserGroups: loadGroups,
+  saveUserGroup: saveGroup,
+  deleteUserGroup: deleteGroup,
+}: UserGroupProps) {
+  const [sidePanelOpened, setSidePanelOpened] = useState(false);
+  const [selectedGroup, setSelectedGroup] = useState<UserGroupSchema>();
+  const groupList = userGroups ? userGroups.toJS() : [];
+  const userList = users ? users.toJS() : [];
+
+  useEffect(() => {
+    if (userGroups === undefined || userGroups.size === 0) {
+      loadGroups().then();
```

Comment:
I still didn't try the functionality so this comment may be useless: Maybe the callback to the `then` is a good place to handle a loading spinner or placeholder.